src/crimson/common/errorator.h

```diff
@@ -596,14 +596,14 @@ struct errorator {
     }
 
     template <class FuncT>
-    auto finally(FuncT &&func) {
+    auto finally(FuncT &&func) -> std::decay_t<decltype(*this)> {
```

Comment:
@athanatos i pulled this commit and revised it in #38305. two changes

- use `_future` as the returned type for better readability
- indent the `discard_result()` line with spaces for better readability in GitHub web interface and for better consistency with the rest of this source file.

Reply:
because this change breaks the build without #38305, i want to get the latter merged first. hence included it in my PR so we can verify that my change does address the FTBFS along with this `finally` fix.

Reply:
rebased and repushed